tests/wallet/test_coin_selection.py

```diff
@@ -108,7 +108,28 @@ async def test_coin_selection_with_dust(self, a_hash: bytes32) -> None:
             )
             assert result is not None
             assert sum([coin.amount for coin in result]) >= target_amount
-            assert len(result) <= 500
+            assert len(result) == 1  # only one coin should be selected
+
+        for i in range(100):
+            coin_list.append(
+                WalletCoinRecord(
+                    Coin(a_hash, std_hash(i), uint64(2000)), uint32(1), uint32(1), False, True, WalletType(0), 1
+                )
+            )
+        spendable_amount = uint128(spendable_amount + 2000 * 100)
+        for target_amount in [50000, 10000]:  # select the first 100 values
```

Comment:
Try some other targets that don't make us choose exact amount as well

Reply:
gotcha.